### PR TITLE
Alerts: DownloadRecentAlertsToCsv

### PR DESCRIPTION
**Pull request by smunir-onms**
## Jira link(s)
https://opennms.atlassian.net/browse/LOK-2230

## Checklist
* [ ] Follows Lōkahi's [development guidelines.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Development-Guidelines)
* [ ] Appropriate reviewer(s) have been selected.
* [ ] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS-Cloud/lokahi/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.
* [ ] Notify devops of changes to the Charts.
* [ ] Notify documentation team of any changes to names of screens or features (affects URLs).

### CODE REVIEW

File: rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlertService.java [hunk 1]
```diff
@@ -201,4 +213,53 @@ public Mono<ListAlertResponse> getRecentAlertsByNode(
...
+        }
+        if (downloadFormat.equals(DownloadFormat.CSV)) {
+            StringBuilder csvData = new StringBuilder();
+            var csvformat = CSVFormat.Builder.create()
+                    .setHeader("Location", "Type", "Severity", "Description")
+                    .build();
+
+            try (CSVPrinter csvPrinter = new CSVPrinter(csvData, csvformat)) {
+                for (Alert alert : alertList) {
+                    csvPrinter.printRecord(
```
Comment (cgorantla) [inline rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlertService.java:254]: follow the UI for the list of things in CSV :  node name, severity, description, lastUpdated (Date format just like UI shows), acknowledge/unacknowledged

Comment (smunir-onms) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlertService.java:255]: Type changed to Alert Type, Date and Time added as per UI:
Screenshot from 2024-03-08 21-47-03

Comment (cgorantla) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlertService.java:255]: > Type changed to Alert Type, Date and Time added as per UI: Screenshot from 2024-03-08 21-47-03

UX need to update fields, it is missing some required fields, we can discuss this internally

Comment (cgorantla) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlertService.java:262]: Don't think we should even separate Date and Time.

Comment (smunir-onms) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlertService.java:255]: fields updated as discussed.

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**